tests/lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,1232 @@
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) { a = b }",
+            output: "a &&= b;",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) { a = b; }\nfn();",
+            output: "a &&= b;\nfn();",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) { a = b }\nfn();",
+            output: "a &&= b;\nfn();",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) { a = b } fn();",
+            output: "a &&= b; fn();",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) { a = b\n} fn();",
+            output: "a &&= b; fn();",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Spacing
+        {
+            code: "if (a) a  =  b",
+            output: "a  &&=  b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a)\n a = b",
+            output: "a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) {\n a = b; \n}",
+            output: "a &&= b;",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Comments
+        {
+            code: "/* before */ if (a) a = b",
+            output: "/* before */ a &&= b",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) a = b /* after */",
+            output: "a &&= b /* after */",
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) /* between */ a = b",
+            output: null,
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        }, {
+            code: "if (a) a = /* between */ b",
+            output: null,
+            options: ["always", { enforceForIfStatements: true }],
+            errors: [{ messageId: "if", type: "IfStatement" }]
+        },
+
+        // > Members
+        {
+            code: "if (a.b) a.b = c",
+            output: null,
```

Comment:
This can be autofixed. `a.b` is read exactly once and then conditionally assigned a value, so this seems equivalent to `a.b &&= c`.

What shouldn't be autofixed is, for example:

* `if (a.b.c) a.b.c = d` (this might be triggering `a.b` twice)
* `with (unrelated) if (a.b) a.b = c` (the same for `unrelated.a`)